### ISSUE TITLE
Which API to use for latest deployments

### ISSUE THREAD

**Issue by tomkuijsten** [MicrosoftDocs/vsts-rest-api-specs#148]
I'm using the (deployments list)[https://docs.microsoft.com/en-us/rest/api/azure/devops/release/deployments/list?view=azure-devops-rest-5.0] API but it seems a bit limited. I want to create an overview of all latest successful deployments for a certain environment (eg production) for each release definition. The API returns ALL deployments ever done, which is a huge amount in our case.

I know about certain parameters:
- latestAttemptsOnly - will only reduce the number of returned deployments of the **same** release
- deploymentStatus - this one helps, only successful deployments
- minStartedTime - doesn't help, some deployments are done months ago

Can anyone tell me what the most data efficient API call is to achieve this?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a2f84371-bba9-2e93-8386-c84ffc87d26c
* Version Independent ID: 06106c51-b396-9d97-9566-72e7a27375eb
* Content: [Release REST API for Azure DevOps Services - Azure DevOps Services REST API](https://docs.microsoft.com/en-us/rest/api/azure/devops/release/?view=azure-devops-rest-5.1#feedback)
* Content Source: [docs-ref-conceptual/5.1/release/index.md](https://github.com/MicrosoftDocs/vsts-docs-rest-apis/blob/live/docs-ref-conceptual/5.1/release/index.md)
* Service: **vsts**
* Product: **devops**
* GitHub Login: @wnjenkin
* Microsoft Alias: **elbatk**

**Comment by nschonni** [comment]
You might have a different reason for want to look at information, but this existing extension for dashboards if that's your use case https://marketplace.visualstudio.com/items?itemName=ms-devlabs.TeamProjectHealth

**Comment by tomkuijsten** [comment]
We already have that extension, but is not what we're looking for. I want to create a "excel sheet" like:

Product | Dev | Prod
-------- | ----- | ----- 
Product A | 1.10.1-beta1 | 1.10.1
Product B | 2.1.0-alpha2 | 2.0.0

[See example here](https://pasteboard.co/HW0zkPu.jpg)

**Comment by deepankumaresan** [comment]
https://vsrm.dev.azure.com/{organization}/{project}/_apis/release/deployments?api-version=5.0&deploymentStatus=succeeded&latestAttemptsOnly=true&definitionEnvironmentId={definitionEnvironmentId}&$top=1

{definitionEnvironmentId} is unique for Release pipeline and Environment combo that you might need to from https://vsrm.dev.azure.com/{organization}/{project}/_apis/release/deployments?api-version=5.0

**Comment by tomkuijsten** [comment]
That's the one I'm using now. During configuration I build an array of definitionEnvironmentId's and for each id, I do an API call. In our case, hundreds of API call's are done. It's the best solution I've come up with so far.

**Comment by chelnak** [comment]
@tomkuijsten Did you get anywhere with this? Currently trying to do something similar but looking at duration rather than versions.

The deployments endpoint docs are still light..

**Comment by tomkuijsten** [comment]
@chelnak nope, at the time I was busy with the issue, there was not a better solution. Might be something changed, but I didn't check lately.

**Comment by chelnak** [comment]
Alright cool. I’m working on something now that fits my use case. I’ll post
when complete just in case it helps anyone.

I’d really like to see this data in the analytics odata api!

**Comment by edgariscoding** [comment]
@tomkuijsten did you ever get this figured out? 

I can't even get the `latestAttemptsOnly` parameter to work. It returns the same number of deployments whether I set that to true or not.

**Comment by tomkuijsten** [comment]
Nope, never got it to work properly. Not using it for a long time though, might be something fixed in the meantime.
